### ISSUE TITLE
dplug-build error: pluginname.so: No such file or directory

### ISSUE THREAD

**Issue by abaga129** [AuburnSounds/Dplug#323]
I cloned v8.2.3 and built `dplug-build`  I get this error though. The plugins are being built as `.dylib` but dplug-build seems to be looking for `.so`

Am I doing something wrong?

**Comment by 0xEAB** [comment]
> I see now that its because its being prepped to build an installer.

bad (read: confusing) error message in this case anyway

**Comment by p0nce** [comment]
Which OS? which `ld` version? Which  format?

> I see now that its because its being prepped to build an installer. Is it possible to still build macOS plugins the old way?

Yes, don't provide the `--installer` switch.

**Comment by abaga129** [comment]
I'm running ld64-302.3.  Trying to build 64 bit AU on MacOS 10.12.6.
The command I'm running is `dplug-build -c AU` and I get the error.

**Comment by p0nce** [comment]
oh we used to call `dub describe` for dubOutputFileName but don't anymore.
I can see why this create problems.

Dub changed the output file name https://github.com/ohdatboi/dub/commit/ec2b6e35feab9f8e539be554c5cc9a515a3be10f

So if we are to support all DUB version, it's either dub describe, or guess the output filename from a list of potential candidates with a file-exist test (preferred?)

**Comment by abaga129** [comment]
Ah I see. Good catch. That explains why downgrading to older versions of dplug-build doesnt make a difference. I thought I was losing my mind.

I would think the `dub describe` route would be the best since it is able to adapt to possible changes in the future.

**Comment by p0nce** [comment]
`dub describe` would slow down every build a lot by requiring dep resolution. I think we can have it all with:

One solution could be that `dubOutputFileName` is lazy and on first call check which file exist amongst a list of candidates (for mac: libXXX.so, libXXX.dylib, [<dubtargetpath>/libXXX.so, <dubtargetpath>/libXXX.dylib])

**Comment by p0nce** [comment]
As a workaround for now just change `.so` to `.dylib` i'll devise a solution that works for all DUB versions.

**Comment by abaga129** [comment]
Okay thanks for the quick response :)

Couldnt we just mirror dub and add
```D
version(OSX)
    outputFilename = name ~ ".dylib";
```

**Comment by p0nce** [comment]
@abaga129 exactly :)

**Comment by p0nce** [comment]
Yes about the error message, so that next time DUB changes we are warned what to modify.

**Comment by p0nce** [comment]
@abaga129 does 2d5d3c3e3b2cf48ee2cf8ac802b58705413742d1 fix your problem? I haven't tested on Mac yet.

**Comment by p0nce** [comment]
I think ideally you should be able to use any DUB from last 2 years. It's been some time since I worried about which dub version I should use, there used to be important difference.

Nowadays Dplug should support whichever comes with your first compiler in PATH (both LDC and DMD comes with a DUB, sometimes you can have several ones)
